test/OpenTelemetry.AotCompatibility.TestApp/Program.cs

```diff
@@ -14,4 +14,17 @@
 // limitations under the License.
 // </copyright>
 
-Console.WriteLine("Hello, World!");
+using OpenTelemetry.AotCompatibility.TestApp;
+
+try
+{
+    PropertyFetcherAotTest.Test();
+}
+catch (Exception ex)
+{
+    Console.WriteLine(ex);
+    return -1;
+}
+
+Console.WriteLine("Passed.");
+return 0;
```

Comment:
When/how does this get executed and checked to ensure that `0` was returned? Should we run the executable as part of "assert-publish-aot-testApp-warningCount.ps1"?

Reply:
Yes we should.
My bad - when I was adding this code we didn't have the CI leg yet and I didn't want to mess with the test knowing it will change.

@Yun-Ting could you please add this - just run the produced binary and check that it's exit code is 0. It should print out any failures to output already.

Reply:
I've created a side PR for this:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4743